### PR TITLE
CWF-129 (fixed rebase #2) related to changes for updating integration tests

### PR DESCRIPTION
**Pull request by johnbe11**

### CODE REVIEW

**Comment by johnbe11** [comment]
@Allda please review

File: ext/vulnsrc/redhat/redhat.go [hunk 1]
```diff
@@ -31,40 +32,47 @@ import (
...
+var SupportedArches = map[string]bool{"x86_64": true, "noarch": true}
 
 // SupportedDefinitionTypes - supported definition classes
-var SupportedDefinitionTypes = map[string]bool { "patch":true }
+var SupportedDefinitionTypes = map[string]bool{"patch": true}
+
+// pendingVulnNames - quick running reference (by name) of all the vulnerabilities which have been added to the pending list
+var pendingVulnNames = map[string]bool{}
+
+var accumulatedVulnerabilities = []database.VulnerabilityWithAffected{}
```
Comment (Allda) [inline ext/vulnsrc/redhat/redhat.go]: Why do you keep this on a global level? Having a global variable that can be huge is a very bad practice and this use case can be solved by just using local variables and do the merge in `Update()` instead in `CollectVulnsForAdvisory()`

Comment (johnbe11) [reply ext/vulnsrc/redhat/redhat.go]: @Allda that is a good point, let me review the vuln-feature merge strategy ([lines 280-290](https://github.com/johnbe11/clair/blob/CWF-129_fix-rebase-2_updateintegration_tests/ext/vulnsrc/redhat/redhat.go#L280)) and find a better way to handle that.

**Comment by Allda** [comment]
@johnbe11 The code looks good to me now and it can be merged, but before I merge it can you please run your comparison test and verify a diff between vmass and oval v2? When I look into the production database and compare it with your results I still see a difference.
```
select count(*) from vulnerability_affected_feature as vaf;
# vmass: 688887
# oval: 823113

select count(*) from vulnerability as v;
# vmass: 73340
# oval: 63395
```

**Comment by johnbe11** [comment]
@Allda I pushed a copy of the [comparison results](https://github.com/johnbe11/oval-vmaas-compare/blob/master/results/details-missing-from-oval-results.txt) and [script](https://github.com/johnbe11/oval-vmaas-compare/blob/master/scripts/vuln-diffs.sh) up to https://github.com/johnbe11/oval-vmaas-compare/blob/master/

From the comparison, it looks like the majority of discrepancies are from differences in the feeds, where a given advisory in the vmaas feed doesn't appear in the oval feed (e.g.: RHSA-2006:0101).  [Three advisories are present in both](https://github.com/johnbe11/oval-vmaas-compare/blob/master/results/summary-missing-from-oval-results.txt
) but weren't picked up by the oval plugin, I'm investigating those to determine why they're not getting picked up

**Comment by johnbe11** [comment]
@Allda  I've pushed a [set of feature comparisons for 500 vulnerabilities](https://github.com/johnbe11/oval-vmaas-compare/blob/master/results/partial_list_diff_features_diff-vmaas-oval.txt), it looks like the large majority of the discrepancies on the oval side are from *debug packages not being included in the oval feed's advisories.  I'll post the full comparison set in the morning.

**Comment by johnbe11** [comment]
@Allda here's the full set:  https://github.com/johnbe11/oval-vmaas-compare/blob/master/results/list_diff_features_diff-vmaas-oval.json

**Pull request merged**